lib/express_validator.js

```diff
@@ -1,262 +1,335 @@
+var additonalSanitizers = ['trim', 'ltrim', 'rtrim', 'escape', 'stripLow', 'whitelist', 'blacklist', 'normalizeEmail'];
+
+/**
+ * Initalizes a chain of validators
  *
- *    Each call to `check()` will throw an exception by default. To
- *    specify a custom err handler, use `req.onValidationError(errback)`
- *    where errback receives a parameter containing the error message
+ * @class
+ * @param  {(string|string[])}  param         path to property to validate
+ * @param  {string}             failMsg       validation failure message
+ * @param  {Request}            req           request to attach validation errors
+ * @param  {string}             location      request property to find value (body, params, query, etc.)
+ * @param  {object}             options       options containing error formatter
+ */
+
+var ValidatorChain = function(param, failMsg, req, location, options) {
+  this.errorFormatter = options.errorFormatter;
+  this.param = param;
+  this.value = (location) ? get(req[location], param) : undefined;
+  this.validationErrors = [];
+  this.failMsg = failMsg;
+  this.req = req;
+
+  return this;
+};
+
+
+/**
+ * Initializes a chain of sanitizers
  *
- * 2. To sanitize parameters, use `req.sanitize(param_name)`
- *        e.g. req.sanitize('param1').toBoolean();
- *        e.g. req.sanitize('param2').toInt();
+ * @class
+ * @param  {(string|string[])}  param    path to property to sanitize
+ * @param  {[type]}             req             request to sanitize
+ * @param  {[type]}             location        request property to find value
+ */
+
+var SanitizerChain = function(param, req, locations) {
+  this.values = locations.map(function(location) {
+    return get(req[location], param);
+  });
+  this.req = req;
+  this.param = param;
+  this.locations = locations;
+  return this;
+};
+
+/**
+ * end sanitizer chain and return sanitized value
  *
- * 3. Done! Access your validated and sanitized paramaters through the
- *    `req.params` object
+ * @method result
+ * @return {string} sanitized value
  */
 
-var validator = require('validator');
+SanitizerChain.prototype.result = function() {
+  return this.values[0];
+};
+
+/**
+ * Adds validation methods to request object via express middleware
+ *
+ * @method expressValidator
+ * @param  {object}         options
+ * @return {function}       middleware
+ */
 
 var expressValidator = function(options) {
-  options = options || {};
+  var defaults = {
+    customValidators: {},
+    customSanitizers: {},
+    errorFormatter: function(param, msg, value) {
+      return {
+        param : param,
+        msg   : msg,
+        value : value
+      };
+    }
+  };
 
-  var _options = {};
+  options = extend(defaults, options);
 
-  _options.customValidators = options.customValidators || {};
-  _options.customSanitizers = options.customSanitizers || {};
+  // set validators and sainitzers as prototype methods on corresponding chains
+  Object.keys(validator).forEach(function(methodName) {
```

Comment:
Might be a good point to introduce the underscore library to be able to write:

_.each(validator, function(validatorMethod, methodName